skimage/filters/_multiotsu.pyx

```diff
@@ -0,0 +1,54 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+import numpy as np
+
+cimport numpy as cnp
+cimport cython
+from itertools import combinations
+
+def _find_threshold_multiotsu(double [:, ::1] var_btwcls,
+                              Py_ssize_t classes,
+                              Py_ssize_t bins,
+                              double [::1] aux_thresh):
+    """
+    Cython utility function for finding thresholds in multi-Otsu algorithm.
+    This function is only called by filters.threshold_multiotsu.
+
+    Parameters
+    ----------
+    var_btwcls : 2-d array
+        array of variance between classes
+    classes : int
+        number of classes to be segmented
+    bins : int
+        number of bins used in the histogram
+    aux_thresh : array
+        thresholds to be returned
+    """
+    cdef Py_ssize_t idd
+    cdef Py_ssize_t sh0, sh1
+    cdef double part_sigma = 0
+    # max_sigma is the maximum variance between classes.
+    cdef double max_sigma = 0
+    cdef Py_ssize_t [:, ::1] _tmp = np.array(list(
+                                combinations(range(1, bins - 2), classes - 1)))
+    sh0 = _tmp.shape[0]
+    sh1 = _tmp.shape[1]
+    cdef Py_ssize_t [:, :] idx_tuples = np.zeros((sh0, sh1 + 2), 
```

Comment:
@alexandrejaguar I wonder if the issue is actually here rather than in the function call (where the types are usually cast implicitly). Try setting the dtype to `Py_ssize_t` or `np.intp_t`?

But I think @stefanv has a better grasp on the different C types and where mismatches might occur...

Reply:
Hey @jni, thanks for your help!
I don't know what's happening there actually, know nothing of cython! :smile: Which dtype should I change? If you point where, I'd appreciate as well 😊

Reply:
it should be `np.intp` i think

Reply:
might be a little more than that. Let me try to have a go at it.

Reply:
I think i'm hitting this on my local machine
https://bugs.python.org/issue32259